Modix.Common/ErrorHandling/ServiceResult.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Modix.Common.ErrorHandling
+{
+    /// <summary>
+    /// Represents the result of a service call with no return value
+    /// </summary>
+    public class ServiceResult
+    {
+        public bool IsSuccess { get; protected set; }
+        public virtual bool IsFailure => !IsSuccess;
+        public string Error { get; protected set; }
+
+        protected ServiceResult() { }
+
+        /// <summary>
+        /// Returns the ServiceResult as an exception for throwing
+        /// </summary>
+        public virtual Exception AsException()
+        {
+            return new InvalidOperationException(Error);
+        }
+
+        public override string ToString()
+        {
+            if (IsSuccess)
+            {
+                return "Success";
+            }
+
+            return $"Failure: {Error}";
+        }
+
+        /// <summary>
+        /// Returns a successful ServiceResult
+        /// </summary>
+        public static ServiceResult FromSuccess()
```

Comment:
Would there be any disadvantage to making this a singleton? Like, the idea is that all `ServiceResult` objects are immutable, yeah? Save us some memory allocations and just use the same `Success` result wherever we need one.

Reply:
Probably fine, yeah. Done.